packages/cli-kit/src/public/node/ruby.test.ts

```diff
@@ -47,138 +54,176 @@ describe('execCLI', () => {
+    await file.inTemporaryDirectory(async (cli2Directory) => {
+      // Given
+      const existingWindowsDependency = true
+      const execSpy = await mockEmbeddedCLI2(cli2Directory, isWindows, !existingWindowsDependency)
+      const gemfilePath = joinPath(cli2Directory, 'Gemfile')
+
+      // When
+      await execCLI2(['args'], {
+        token: 'token_0000_1111_2222_3333',
+        directory: './directory',
+      })
+
+      // Then
+      validateBudleExec(execSpy, gemfilePath, joinPath(cli2Directory, 'bin', 'shopify'))
+      await validateGemFileContent(gemfilePath, !isBundled, isWindows)
     })
+  })
 
-    // Then
-    expect(execSpy).toHaveBeenLastCalledWith('bundle', ['exec', '/embed/internal/bin/shopify', 'args'], {
-      stdio: 'inherit',
-      cwd: './directory',
-      env: {
-        ...process.env,
-        SHOPIFY_CLI_STOREFRONT_RENDERER_AUTH_TOKEN: undefined,
-        SHOPIFY_CLI_ADMIN_AUTH_TOKEN: undefined,
-        SHOPIFY_SHOP: undefined,
-        SHOPIFY_CLI_AUTH_TOKEN: 'token_0000_1111_2222_3333',
-        SHOPIFY_CLI_RUN_AS_SUBPROCESS: 'true',
-        BUNDLE_GEMFILE: '/embed/internal/Gemfile',
-      },
-      signal: undefined,
+  it('when run embedded CLI2 in windows with existing dependency then gemfile content should be correct and bundle runs with correct params', async () => {
+    await file.inTemporaryDirectory(async (cli2Directory) => {
+      // Given
+      const existingWindowsDependency = true
+      const execSpy = await mockEmbeddedCLI2(cli2Directory, isWindows, existingWindowsDependency)
+      const gemfilePath = joinPath(cli2Directory, 'Gemfile')
+
+      // When
+      await execCLI2(['args'], {
+        token: 'token_0000_1111_2222_3333',
+        directory: './directory',
+      })
+
+      // Then
+      validateBudleExec(execSpy, gemfilePath, joinPath(cli2Directory, 'bin', 'shopify'))
+      await validateGemFileContent(gemfilePath, !isBundled, isWindows)
     })
-
-    // Teardown
-    process.env = originalEnv
   })
 })
+
+function mockBundledCLI2(cli2Directory: string, windows: boolean) {
+  vi.spyOn(local, 'isShopify').mockResolvedValue(false)
+  vi.spyOn(pathConstants.directories.cache.vendor, 'path').mockReturnValue(cli2Directory)
+  mockRubyEnvironment()
+  mockPlatformAndArch(windows)
+  return vi.spyOn(system, 'exec')
+}
+
+async function mockEmbeddedCLI2(cli2Directory: string, windows: boolean, existingWindowsDependency = false) {
+  vi.spyOn(local, 'isShopify').mockResolvedValue(true)
+  vi.spyOn(file, 'findPathUp').mockResolvedValue(cli2Directory)
+  mockRubyEnvironment()
+  mockPlatformAndArch(windows)
+  await createGemFile(cli2Directory, existingWindowsDependency)
+  return vi.spyOn(system, 'exec')
+}
+
+function mockRubyEnvironment() {
+  vi.mocked(captureOutput).mockResolvedValueOnce('2.7.5')
+  vi.mocked(captureOutput).mockResolvedValueOnce('2.4.0')
+}
+
+function mockPlatformAndArch(windows: boolean) {
```

Comment:
The following piece of code will make the code hard to understand:

```js
mockPlatformAndArch(true)
```

What about:

```js
mockPlatformAndArch({ isWindows: true })
```

Reply:
yes, it's the same approach mentioned [here](https://github.com/Shopify/cli/pull/1350#discussion_r1109484229)